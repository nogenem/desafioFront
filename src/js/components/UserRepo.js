const renderUserRepo = ({
  html_url,
  name,
  language,
  description,
  subscribers_count,
  stargazers_count,
  forks_count
}) => `
  <div id="repoCard" class="row align-items-center justify-content-center">
    <div class="card col-6 text-center">
      <div class="card-body">
        <div class="row">
          <div class="col-12">
            <h3 class="mb-0 text-truncated">
              <a href="${html_url}" target="_blank">${name}</a>
            </h3>
            <p class="lead">${language}</p>
            <p>
              ${description || ""}
            </p>
          </div>
          <div class="col-12 col-lg-4">
            <h3 class="mb-0">${subscribers_count}</h3>
            <small>Assinante(s)</small>
          </div>
          <div class="col-12 col-lg-4">
            <h3 class="mb-0">${stargazers_count}</h3>
            <small>Stars</small>
          </div>
          <div class="col-12 col-lg-4">
            <h3 class="mb-0">${forks_count}</h3>
            <small>Forks</small>
          </div>
        </div>
      </div>
    </div>
  </div>
`;

export default {
  render: renderUserRepo
};
