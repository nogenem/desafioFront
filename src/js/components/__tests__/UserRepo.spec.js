import UserRepo from "../UserRepo";

const testData = {
  html_url: "http://github.com/test/testProj",
  name: "testProj",
  language: "JavaScript",
  description: "Some proj",
  subscribers_count: "0",
  stargazers_count: "0",
  forks_count: "0"
};

it("renders correctly when passed all props", () => {
  const output = UserRepo.render(testData);
  expect(output).toMatchSnapshot();
});

it("renders correctly when not passed all props", () => {
  const output = UserRepo.render({
    ...testData,
    description: undefined
  });
  expect(output).toMatchSnapshot();
});
