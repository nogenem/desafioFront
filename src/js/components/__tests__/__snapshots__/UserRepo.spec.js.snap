// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when not passed all props 1`] = `
"
  <div id=\\"repoCard\\" class=\\"row align-items-center justify-content-center\\">
    <div class=\\"card col-6 text-center\\">
      <div class=\\"card-body\\">
        <div class=\\"row\\">
          <div class=\\"col-12\\">
            <h3 class=\\"mb-0 text-truncated\\">
              <a href=\\"http://github.com/test/testProj\\" target=\\"_blank\\">testProj</a>
            </h3>
            <p class=\\"lead\\">JavaScript</p>
            <p>
              
            </p>
          </div>
          <div class=\\"col-12 col-lg-4\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Assinante(s)</small>
          </div>
          <div class=\\"col-12 col-lg-4\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Stars</small>
          </div>
          <div class=\\"col-12 col-lg-4\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Forks</small>
          </div>
        </div>
      </div>
    </div>
  </div>
"
`;

exports[`renders correctly when passed all props 1`] = `
"
  <div id=\\"repoCard\\" class=\\"row align-items-center justify-content-center\\">
    <div class=\\"card col-6 text-center\\">
      <div class=\\"card-body\\">
        <div class=\\"row\\">
          <div class=\\"col-12\\">
            <h3 class=\\"mb-0 text-truncated\\">
              <a href=\\"http://github.com/test/testProj\\" target=\\"_blank\\">testProj</a>
            </h3>
            <p class=\\"lead\\">JavaScript</p>
            <p>
              Some proj
            </p>
          </div>
          <div class=\\"col-12 col-lg-4\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Assinante(s)</small>
          </div>
          <div class=\\"col-12 col-lg-4\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Stars</small>
          </div>
          <div class=\\"col-12 col-lg-4\\">
            <h3 class=\\"mb-0\\">0</h3>
            <small>Forks</small>
          </div>
        </div>
      </div>
    </div>
  </div>
"
`;
